src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -55,22 +96,36 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
 
-    deinitialize(): Promise<InternalCheckoutSelectors> {
-        this._paypalCommercePaymentProcessor.deinitialize();
+    async deinitialize(): Promise<InternalCheckoutSelectors> {
+        if (this._isPaypalButton) {
+            if (!this._paymentMethod) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            await this._store.dispatch(this._paymentStrategyActionCreator.disableEmbeddedSubmitButton(this._paymentMethod.id));
+        }
+
+        this._paymentMethod = undefined;
+        this._orderId = undefined;
+        this._isPaypalButton = false;
 
         return Promise.resolve(this._store.getState());
     }
 
-    private async _getOrderId(methodId: string): Promise<string> {
-        const state = this._store.getState();
-        const cart = state.cart.getCartOrThrow();
-        const provider = methodId === 'paypalcommercecredit' ? 'paypalcommercecreditcheckout' : 'paypalcommercecheckout';
-        const { approveUrl, orderId } = await this._paypalCommerceRequestSender.setupPayment(provider, cart.id);
```

Comment:
thanks for the suggestion @davidchin, there is an option to send ajax call in advance, however, initially, we have discussed with PayPal an alternative solution for opening the popup via native PayPal SDK, because `window.open` solution is tied to events that PayPal does not guaratee to be working in future (for popup closing) https://github.com/bigcommerce/checkout-sdk-js/pull/980/files#diff-3d44283948e4a50fdfa4cefc761563bfL59 

```js
 const messageHandler = (event: MessageEvent) => {
    if (event.origin !== 'https://www.sandbox.paypal.com' && event.origin !== 'https://www.paypal.com') {
       return;
    }

   const data = JSON.parse(event.data);

   if (data.operation === 'return_to_merchant' && data.updateParent) {
        this._window.removeEventListener('message', messageHandler);
        closeWindow(true);
   }
};
this._window.addEventListener('message', messageHandler);
```
So we decided to move on with the native PayPal buttons solution, as in that case there is a guarantee so it can work in the future + PayPal maintains popup solution for different browsers

Reply:
I'm not sure that it's good point. We decided to implement PayPal smart button to checkout. 
With sbp we shouldn't worry about browser support, postMessages, changes on the PayPal side (with `window.open` we're waiting message with `operation` and `updateParent` and if PayPal change it - it wouldn't work in our side)
And now it work well